django/bosscore/views/views_resource.py

```diff
@@ -323,15 +320,9 @@ def get(self, request, collection, experiment):
             collection_obj = Collection.objects.get(name=collection)
             experiment_obj = Experiment.objects.get(name=experiment, collection=collection_obj)
             # Check for permissions
-            # TODO SH Hack added to allow us to quickly make experiment detail public without logging in.
             if experiment_obj is None:
                 return BossResourceNotFoundError(experiment)
-            if experiment_obj.id in [67, 68, 76, 77, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94,
-                                     95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111,
-                                     112, 113, 114, 115, 116, 117, 118, 120, 121, 122, 123, 124, 125, 126, 127, 128,
-                                     129, 130, 133, 134, 139, 140, 141, 142, 143, 147, 148, 155, 156, 158, 160, 162,
-                                     163, 164, 165, 166, 167, 168, 169, 170, 176, 177, 180, 181, 182, 183, 184, 185,
-                                     186] or request.user.has_perm("read", experiment_obj):
+            if experiment_obj.public or request.user.has_perm("read", experiment_obj):
```

Comment:
A quick question to make sure I'm understanding,

setup:

* my collection is public
* my experiment is public
* `channelA` is public, `channelB` is private

what do I see if I ask for exp info? do I see both channels (anon user)?

Reply:
Good question! Definitely want to only show the public one only but need to put a test in place to prove that works. I forgot about this which is why I added you to review!